src/diffusers/models/attention.py

```diff
@@ -120,13 +120,17 @@ def __init__(
         norm_type: str = "layer_norm",
         final_dropout: bool = False,
         attention_type: str = "default",
+        caption_channels: int = None,
```

Comment:
`caption_channels` is supposed to be an `int` here, but it's never used anywhere (as an int). It's only used as:

```py
is not None
```

Can we instead just create a new norm type?

Reply:
Let's delete the input `caption_channels` completely and instead add a new `norm_type="ada_layer_norm_single"`